### PR TITLE
Make the build more IDEA friendly

### PR DESCRIPTION
**Pull request by chirino**
With these changes you should be able to run the embedded smoke tests even after an IDEA rebuild.

### CODE REVIEW

File: fabric/spi/src/main/java/io/fabric8/spi/internal/BootConfigurationImpl.java [hunk 1]
```diff
@@ -77,4 +77,11 @@ public Version getVersion() {
...
         return Collections.unmodifiableMap(configuration);
     }
+
+    protected void bindConfigurer(Configurer configurer){
+        this.configurer=configurer;
+    }
+    protected void unbindConfigurer(Configurer configurer) {
+        this.configurer=null;
+    }
 }
```
Comment (tdiesler) [inline fabric/spi/src/main/java/io/fabric8/spi/internal/BootConfigurationImpl.java:87]: why would these need to be protected? we do not extend components

Comment (chirino) [inline fabric/spi/src/main/java/io/fabric8/spi/internal/BootConfigurationImpl.java:87]: The maven-scr-plugin was generating these methods for us.  It was generating them with the protected scope.  I just added the methods to match.

Comment (tdiesler) [inline fabric/spi/src/main/java/io/fabric8/spi/internal/BootConfigurationImpl.java:87]: We use ValidatingReference for a number of reasons - one of them is that a reference can only be set to null when the old ref is the same as the given unbound ref. 
The code below does not check for that condition. Generally, we want to limit visibility - it documents its possible accessibility.

If you see bind/unbind (or other scr lifecycle methods) in scope protected/public they should be changed to default scope 

cheers
—thomas

**Comment by tdiesler** [comment]
merged

**Pull request closed without merging**